Add proposal for the deals.lose endpoint

WouterSioen opened the pull request:
It allows you to optionally add a loss reason and an elaboration (just
like in the UI).

previous proposal was `.lost` instead of `.lose`, but we thought `.lose` is more consistent with other endpoints (actions in the imperative form). Open for discussion.

stivni commented on src/general.apib:
```diff
@@ -14,7 +14,8 @@ Get a list of all users
                 + gsm: `+32412654321` (string)
                 + phone: `+3291239843` (string)
-                + team_id: `1` (number)
+                + team: (object)
```
but should we use `teams : (array)` for the future?

stivni commented on src/companies.apib:
```diff
@@ -20,7 +20,26 @@ Search companies
+                + bic: `BICBANK` (string)
+                + language: `NL` (string)
+                + tags (object)
+                    + _items : tag1, tag2 (array[string])
```
Q : would it make sense for out tags to be objects with only a name too?
In that case, we could later include on _actions_ for it... (the untag action for it, for example)

stivni replied:
@teamleadercrm/developers any opinions?

regularjack replied:
It makes sense but I would only implement it when needed

vanbrabantf commented:
Q: doesn't our spec go much further then that?
I know we didn't want to implement it all at once, but wouldn't a selected rollout just cause more confusion?

stivni commented:
@vanbrabantf, yes, it should include types, links, embedded data, included data, ...
but I think we have more important things to tackle first... like relaunch a working stable api

WouterSioen committed:
Add proposal for the deals.lose endpoint

It allows you to optionally add a loss reason and an elaboration (just
like in the UI).

JordyDeGroeve reviewed (approved):
👍

WouterSioen committed:
Merge branch 'master' into CORE-493-deals.lose
